src/main/java/de/tum/in/www1/artemis/web/rest/settings/ide/IdeSettingsResource.java

```diff
@@ -0,0 +1,109 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.domain.settings.ide.Ide;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMapping;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMappingId;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.settings.IdeRepository;
+import de.tum.in.www1.artemis.repository.settings.UserIdeMappingRepository;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+
+/**
+ * REST controller for managing NotificationSettings (NotificationSettings).
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class IdeSettingsResource {
+
+    private static final Logger log = LoggerFactory.getLogger(IdeSettingsResource.class);
+
+    private final UserIdeMappingRepository userIdeMappingRepository;
+
+    private final IdeRepository ideRepository;
+
+    private final UserRepository userRepository;
+
+    public IdeSettingsResource(UserIdeMappingRepository userIdeMappingRepository, IdeRepository ideRepository, UserRepository userRepository) {
+        this.userIdeMappingRepository = userIdeMappingRepository;
+        this.ideRepository = ideRepository;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * GET users/ides: get the predefined IDEs
+     *
+     * @return the ResponseEntity with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     */
+    @GetMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<?> getIdesOfUser() {
+        User user = userRepository.getUser();
+        log.debug("REST request to get IDEs of user {}", user.getLogin());
+
+        var ideMappings = userIdeMappingRepository.findAllByUserId(user.getId());
+        var ideRecords = ideMappings.stream().map(x -> new IdeMappingDTO(x.getProgrammingLanguage(), x.getIde())).toArray();
+        log.debug("Successfully queried IDEs of user {}", user.getLogin());
+
+        return ResponseEntity.ok(ideRecords);
+    }
+
+    /**
+     * PUT users/ides: set the IDE for a programming Language of the user
+     *
+     * @return the ResponseEntity with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     */
+    @PutMapping("ide-settings")
+    @EnforceAtLeastStudent
+    public ResponseEntity<?> setIde(@RequestParam ProgrammingLanguage programmingLanguage, @RequestBody IdeDTO ide) {
+        User user = userRepository.getUser();
+        log.debug("REST request to set IDE of user {}", user.getLogin());
+
+        // find or create the ide
+        var savedIde = ideRepository.findByDeepLink(ide.deepLink()).orElse(null);
+        if (savedIde == null) {
+            savedIde = ideRepository.save(new Ide(ide.name(), ide.deepLink()));
```

Comment:
As you are inserting data into the DB here, you could think of some data sanitization to increase security and usability. e.g., stripping the name and link or verifying that the link is in some format (like `contains ://`)